frontend/src/components/CalendarView/CalendarView.jsx

```diff
@@ -128,18 +131,36 @@ function CalendarView () {
           allDay: info.event.allDay
         }
         const backendUrlAccess = import.meta.env.VITE_BACKEND_ADDRESS;
-        const options = {
-        method: 'PUT',
-        headers: {
-            'Accept': 'application/json',
-            'Content-Type': 'application/json'},
-        body: JSON.stringify(updatedEvent),
-        credentials: 'include'
-          };
-        const response = await fetch(`${backendUrlAccess}/calendar/events/${info.event.id}`, options);
-        if (!response.ok) {
-          throw new Error('Something went wrong!');
+
+        // Handles for Editing Google Calendar Events
+        if (info.event.extendedProps.source === 'google'){
+          const options = {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'},
+            body: JSON.stringify({updatedEventData: updatedEvent}),
+            credentials: 'include'
+              };
+          const response = await fetch(`${backendUrlAccess}/google-cal/update-event/${info.event.extendedProps.masterEventId}`, options);
+          if (!response.ok) {
+            throw new Error('Something went wrong!');
```

Comment:
you should have a better error message than this. Does your backend return an error message?